velox/exec/Merge.cpp

```diff
@@ -123,34 +113,79 @@ BlockingReason Merge::isBlocked(ContinueFuture* future) {
   return BlockingReason::kWaitForProducer;
 }
 
-void Merge::startSources() {
-  VELOX_CHECK_LE(numStartedSources_, sources_.size());
-  // Start the merge source once.
-  if (numStartedSources_ >= sources_.size()) {
-    return;
+bool Merge::isFinished() {
+  return finished_;
+}
+
+void Merge::spill() {
+  if (mergeOutputSpiller_ == nullptr) {
+    VELOX_CHECK(spillConfig_.has_value());
+    mergeOutputSpiller_ = std::make_unique<MergeSpiller>(
+        outputType_,
+        std::nullopt,
+        HashBitRange{},
+        sortingKeys_,
+        &spillConfig_.value(),
+        &spillStats_);
+    hasSpilled_ = true;
   }
-  VELOX_CHECK_EQ(numStartedSources_, 0);
-  VELOX_CHECK(streams_.empty());
-  VELOX_CHECK(sourceBlockingFutures_.empty());
-  // TODO: support lazy start for local merge with a large number of sources
-  // to cap the memory usage.
-  for (auto& source : sources_) {
-    source->start();
+
+  if (output_ != nullptr) {
+    numSpilledRows_ += output_->size();
+    mergeOutputSpiller_->spill(SpillPartitionId{0}, output_);
+    output_ = nullptr;
   }
-  numStartedSources_ = sources_.size();
 }
 
-bool Merge::isFinished() {
-  return finished_;
+void Merge::finishSpill() {
+  SpillPartitionSet spillPartitionSet;
+  mergeOutputSpiller_->finishSpill(spillPartitionSet);
+  mergeOutputSpiller_ = nullptr;
+  VELOX_CHECK_EQ(spillPartitionSet.size(), 1);
+  const auto& spillFiles = spillPartitionSet.cbegin()->second->files();
+  std::vector<std::unique_ptr<SpillReadFile>> spillReadFiles;
```

Comment:
Shall we open spill file until we start processing the spilled file after all the outputs have been consumed?